lib/executor.js

```diff
@@ -0,0 +1,62 @@
+"use strict" ;
+
+module.exports = class Executor {
+    /* Creates a new executor
+    @params [MeasureSource] measure_source -- the source to retrieve the executatble measures from
+    */
+  constructor(measure_source) {
+    this.measure_source = measure_source;
+  }
+
+  /*
+  Generates the executable measures for calculation
+  @params [array] measure_ids -- the measures to generate
+  @params [object] options -- options used to generate the measures such as effective_date, enable_logging ....
+  @returns [array] an array of executable measures
+
+  */
+  get_measures_by_id(measure_ids, options) {
+    var measures = [];
+    // measure_ids parameter will either be a single id or multiple ids.
+    // If the former, transform it to an array
+    measure_ids = Array.isArray(measure_ids) ? measure_ids : [measure_ids];
+    measure_ids.forEach(id => {
+      var mes = this.measure_source.getMeasure(id);
+      measures.push(mes.generate(options));
```

Comment:
@ljades what's your vision for what Measure.generate will do?

Reply:
After the measure is returned, adapting it to work with options given such as effective_date, enable_logging, etc. will allow measure calculation to be done directly on it (with potentially little to no input aside from the patient itself) later.

It's possible that when this was originally planned (with HQMF/QDM) this function would've also generated the Javascript methods from the raw measure, but with CQL that happens at runtime during calculation so there's no need for that.